### ISSUE TITLE
Normative references to a specification of metadata properties are needed.

### ISSUE THREAD

**Issue by jasonjgw** [w3c/epub-specs#1799]
In section 2.2 of EPUB Accessibility 1.1, which is normative, several metadata properties are cited with brief descriptions. For details, one is referred to a non-normative techniques document. There is no normative reference to the allowable types and values of the metadata fields (including those which are required by the EPUB Accessibility specification).

An EPUB document could conform to EPUB Accessibility by using the same metadata property names as specified, but departing from the allowable property values as given in the Techniques document and the non-normative schema.org site to which it refers. This introduces a problem of interoperability resulting from the lack of a suitable normative reference in the specification. That this is so is particularly concerning in view of the observation that some of the metadata are required to be present.

**Comment by mattgarrish** [comment]
Unfortunately, I don't believe we can fix this.

In the original IDPF 1.0 version of this specification we directly referenced the schema.org metadata properties, but schema.org was not considered a stable reference standard when we took the document through ISO standardization.

From recent experience with the Publication Manifest specification, citing schema.org is also not considered stable in W3C, so we couldn't switch back if we wanted to.

I agree this is a frustrating way to standardize the metadata, but I don't know what other choice we have but to describe the properties that have to be expressed. Yes, you could make up your own metadata, but I don't expect it's a problem we'll run into in practice.

Ideally, schema.org will become more acceptable for referencing in the future at which point we could revert back to the way things were.

**Comment by jasonjgw** [comment]
You could specify the allowed property values directly in EPUB Accessibility 1.1 instead of relying on an external reference. In the future, if an external reference becomes available, you could update the specification to cite it. This would give you a stable set of allowed values for purposes of EPUB.

**Comment by mattgarrish** [comment]
> You could specify the allowed property values directly in EPUB Accessibility 1.1

We've had ongoing discussions about how to standardize the values for the schema.org properties because the current web schemas wiki page where they're hosted is even more problematic than schema.org itself.

One possibility is a W3C registry, which could be referenced, but it's complicated by the metadata not belonging to W3C -- it was developed by Benetech from an IMS standard. The more likely scenario is they get maintained in a community group similar to how schema.org is generally maintained, but that leads us back down the path of not being a referenceable standard.

Embedding the values in the specification has similar ownership issues to what the registry option presents.

**Comment by iherman** [comment]
I think we have now a valid precedence, actually, which is the [Publication Manifest](https://www.w3.org/TR/pub-manifest/) specification. As @mattgarrish has referred to, that document references schema.org properties (e.g., [accessibility properties](https://www.w3.org/TR/pub-manifest/#accessibility)), and schema.org is listed as part of the normative references. Ie, schema.org is now acceptable, per [the relevant w3c document](https://www.w3.org/2013/09/normative-references).

**Comment by mattgarrish** [comment]
> that document references schema.org properties

But only as mappings. I thought that was the acceptable workaround to actually using the properties directly for the manifest, or am I forgetting now where we ended up on the issue?

What would be the equivalent solution here: can we name the properties and avoid prose descriptions of them, or do we have to keep the prose descriptions but could "map" them to the schema.org properties instead of waiting for the techniques to do that part?

**Comment by iherman** [comment]
I would ask the director ahead of time, like now. I understand that: 

> In the original IDPF 1.0 version of this specification we directly referenced the schema.org metadata properties, but schema.org was not considered a stable reference standard when we took the document through ISO standardization.

But W3C might be more flexible than ISO.

Is the question whether we can, normatively, refer to the Accessibility properties that are defined as schema.org properties? What is exactly the decision procedure for those properties? I thought the semantics are under W3C control...

**Comment by mattgarrish** [comment]
> Is the question whether we can, normatively, refer to the Accessibility properties that are defined as schema.org properties?

Yes. For example, in 1.0 we listed the property with its definition:

> - accessMode — a human sensory perceptual system or ...

But had to change it to:

> - Access modes — a human sensory perceptual system or ...

To avoid listing the schema.org property that you have to use. Since only schema.org defines equivalent properties that you can use in the package document, we've gotten away with prose descriptions.

> I thought the semantics are under W3C control.

No, the properties were developed in partnership with IMS long before the EPUB accessibility specification came along (for schema.org, and later adapted for EPUB). They're based on IMS' Access for All metadata.

They're under W3C control in the sense that schema.org is run through a community group now, but IMS is still the formal owner of the metadata. That's what complicates these discussions about maintaining both the properties and their values.

**Comment by jasonjgw** [comment]
I think a registry of values would be an effective solution, if it were to come into existence. The clear advantage is that the values could be updated in one place for all purposes.

Specifying the allowable property values explicitly in EPUB Accessibility 1.0 would not, it seems to me, constitute an assertion of "ownership" of the respective properties. It would merely assert (perhaps with an informative reference to the source) a defined subset of property values taken from schema.org as permissible for purposes of describing the accessibility of EPUB documents, without limiting what values may be defined in the future for other purposes - such as general Web page metadata. Of course, this would require EPUB Accessibility to be updated to reflect any changes to these definitions that should be supported for EPUB purposes.

What isn't satisfactory, in my view, is leaving an essential component of the proposed EPUB Accessibility standard unspecified, as the draft currently does. Any solution that provides a normative reference to or specification of the metadata properties would resolve the issue from my perspective.

**Comment by iherman** [comment]
I think that having a separate registry would be a disservice to the community; if the reality is that the community uses the schema.org term that duplicating those would just lead to confusion.

@mattgarrish thanks for the summary in https://github.com/w3c/epub-specs/issues/1799#issuecomment-915165653. I would propose to ask the director explicitly about that. I will put together this question, will cc you and post the reference to this thread.

**Comment by mattgarrish** [comment]
> I think that having a separate registry would be a disservice to the community;

Agree. This rarely works in practice. We tried a similar approach for edupub, as I recall, for the schema.org educational property values and eventually had to warn people to be careful about using the lists because they got out-dated and contradictory.

The schema.org properties already link to the controlled list of values to use with them, so the issue is more how to formalize the connection between this specification and schema.org. If we find a new home for the values in the future, we can update the schema.org properties without having to do anything to this specification.

It would be great it we can go back to referencing the schema.org properties directly, so I've got my fingers crossed that the simplest solution is the only one we'll need here.

**Comment by iherman** [comment]
FYI, I have contacted @plehegar and @swickr on this.

**Comment by gregoriopellegrino** [comment]
Just to add an item: I'd really like this section to be structured like the metadata section in [EPUB 3.3](https://www.w3.org/TR/epub-33/#sec-pkg-metadata) (I don't know if it's too much work for Matt). One important aspect for me that should be brought back I think is cardinality.

For example, for the _accessibilitySummary_ metadata I think it would make sense to indicate that there can be more than one, only if they are in different languages.

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-09-23-epub-a11y#section4) on 2021-09-23

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. Normative references to a specification of metadata properties are needed. 

_See github issue [#1799](https://github.com/w3c/epub-specs/issues/1799), [#1767](https://github.com/w3c/epub-specs/issues/1767)._




_See github pull request [#1818](https://github.com/w3c/epub-specs/pull/1818), [#1815](https://github.com/w3c/epub-specs/pull/1815)._




**Avneesh Singh:** These are the reminders, there are some PRs with requests to review  

**Avneesh Singh:** there's some further discussion on zero-tolerance for a11y review, please provide comments  
… AOB?  

**George Kerscher:** Asking for comments for these on GH?  

**Avneesh Singh:** The PRs need review, and the last one is an issue, please provide comments on that  

**George Kerscher:** Personally, the approach being taken for discovery seems ok  

**Avneesh Singh:** There's more emails  

**Charles LaPierre:** I was just looking at the editor's note FXL techniques  
… diff and preview have errors  

> *Charles LaPierre:* Add editor's note mentioning the fxl techniques #1818

> *Charles LaPierre:* [https://github.com/w3c/epub-specs/pull/1818](https://github.com/w3c/epub-specs/pull/1818)

**Wendy Reid:** It's already merged, so there's no "diff"  

> *John Foliot:* matt, FWIW, another emergent spec with a section related to security and privacy. Note that this WG are in ongoing discussions with i18n folks at W3C. See: [https://w3c.github.io/personalization-semantics/content/index.html#privacy-and-security-considerations](https://w3c.github.io/personalization-semantics/content/index.html#privacy-and-security-considerations)

**Matt Garrish:** Just look at the file changes  

**Avneesh Singh:** Let's continue discussion on GH  
… APA call times will be finalized, and I will send them to the mailing list  
… see you all in the 3.3 call tomorrow  
… have a nice day!  

---


</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-09-30-epub#section3) on 2021-09-30

List of resolutions:

- ***[Resolution No. 2](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-09-30-epub#resolution2):  Merge PR 1808, leave issue 1799 open until schema issue is resolved***

<details><summary><i>View the transcript</i></summary>

### 3. Schema in EPUB Accessibility

_See github issue [#1799](https://github.com/w3c/epub-specs/issues/1799)._




_See github pull request [#1808](https://github.com/w3c/epub-specs/pull/1808)._




**Matt Garrish:** we started off listing the `schema.org` properties, then we went to the ISO version of the document where `schema.org` is not considered very normative, so we referred to the techniques document... so we have to prove that `schema.org` is a stable and valid reference  
… we need to show there is a method for maintaining the vocabularies. Need to work with W3C; it is an ongoing conversation. Still in limbo until we have a proper home for the metadata.  

**Dave Cramer:** can we merge the pull request?  

**Matt Garrish:** would like to get a formal ok first from Ivan  

**Dave Cramer:** we should merge it instead of leaving it hanging as we seek horizontal reviews, etc.  

**Wendy Reid:** we reference schema elsewhere  

**Matt Garrish:** we have a mapping file in those cases  

**Murata Makoto:** can we use living document?  

**Wendy Reid:** there are procedures for using living documents  

**Dave Cramer:** can we propose to merge the PR?  

**Matt Garrish:** it's been referenced in other standards, yet we need to show stability even though it's been allowed elsewhere  
… need a repository and issue tracker to show that we maintain it properly  

**Murata Makoto:** Registries are considered a back door for extending international standards; that's why ISO doesn't like it. Also there are fees associated.  

**Matt Garrish:** it's a gray area. It's part of the web and we have to work with it.  

> *Murata Makoto:* Let's make Ivan's life interesting!

> **Proposed resolution: Merge PR 1808, leave issue 1799 open until schema issue is resolved** *(Wendy Reid)*

> *Dave Cramer:* +1

> *Shinya Takami (高見真也):* +1

> *Murata Makoto:* +1

> *Matt Garrish:* +1

> *Wendy Reid:* +1

> *Toshiaki Koike:* +1

> *Masakazu Kitahara:* +1

> *Dan Lazin:* +1

> *Ben Schroeter:* 0

> ***Resolution #2: Merge PR 1808, leave issue 1799 open until schema issue is resolved***

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section2) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 2. Normative references to a specification of metadata properties are needed: #1799.

_See github issue [epub-specs#1799](https://github.com/w3c/epub-specs/issues/1799)._




> *Avneesh Singh:* new CG for governance: [https://www.w3.org/community/a11y-discov-vocab/](https://www.w3.org/community/a11y-discov-vocab/).

> *Avneesh Singh:* published vocab: [https://www.w3.org/2021/a11y-discov-vocab/latest/](https://www.w3.org/2021/a11y-discov-vocab/latest/).

**Matt Garrish:** basically this issue is about whether we're allowed to use Schema references. We've created CG to maintain vocab that was in Schema.  

> *Avneesh Singh:* GitHub: [https://github.com/w3c/a11y-discov-vocab/blob/main/README.md](https://github.com/w3c/a11y-discov-vocab/blob/main/README.md).

**Matt Garrish:** CG document has been published, and open PR updates the references in our spec and techniques doc.  

**Matt Garrish:** there is an issue open with Schema.org on which we are still waiting for their approval.  
… in the meantime we're holding.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section2) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 2. Normative references to a specification of metadata properties are needed: #1799.

_See github issue [epub-specs#1799](https://github.com/w3c/epub-specs/issues/1799)._




> *Avneesh Singh:* new CG for governance: [https://www.w3.org/community/a11y-discov-vocab/](https://www.w3.org/community/a11y-discov-vocab/).

> *Avneesh Singh:* published vocab: [https://www.w3.org/2021/a11y-discov-vocab/latest/](https://www.w3.org/2021/a11y-discov-vocab/latest/).

**Matt Garrish:** basically this issue is about whether we're allowed to use Schema references. We've created CG to maintain vocab that was in Schema.  

> *Avneesh Singh:* GitHub: [https://github.com/w3c/a11y-discov-vocab/blob/main/README.md](https://github.com/w3c/a11y-discov-vocab/blob/main/README.md).

**Matt Garrish:** CG document has been published, and open PR updates the references in our spec and techniques doc.  

**Matt Garrish:** there is an issue open with Schema.org on which we are still waiting for their approval.  
… in the meantime we're holding.  

</details>

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-11-11-epub-a11y#section2) on 2021-11-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 2. Normative references to a specification of metadata properties are needed: #1799.

_See github issue [epub-specs#1799](https://github.com/w3c/epub-specs/issues/1799)._




> *Avneesh Singh:* new CG for governance: [https://www.w3.org/community/a11y-discov-vocab/](https://www.w3.org/community/a11y-discov-vocab/).

> *Avneesh Singh:* published vocab: [https://www.w3.org/2021/a11y-discov-vocab/latest/](https://www.w3.org/2021/a11y-discov-vocab/latest/).

**Matt Garrish:** basically this issue is about whether we're allowed to use Schema references. We've created CG to maintain vocab that was in Schema.  

> *Avneesh Singh:* GitHub: [https://github.com/w3c/a11y-discov-vocab/blob/main/README.md](https://github.com/w3c/a11y-discov-vocab/blob/main/README.md).

**Matt Garrish:** CG document has been published, and open PR updates the references in our spec and techniques doc.  

**Matt Garrish:** there is an issue open with Schema.org on which we are still waiting for their approval.  
… in the meantime we're holding.  

</details>

**Comment by iherman** [comment]
@mattgarrish where are we with this issue? The document refers to schema.org properties directly (e.g., to https://schema.org/accessibilityFeature) which might be enough for the purposes of W3C.

That being said, if I follow my nose on the schema.org page, and goes to open issues, I get to a [schema issue list](https://github.com/schemaorg/schemaorg/issues?q=is%3Aissue+is%3Aopen+accessibilityFeature) as opposed to [the CG's repository](https://github.com/w3c/a11y-discov-vocab/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc) what is the situation right now v.a.v. schema.org?

If you think that this issue is ready to be closed from the point of view of the EPUB spec (which I believe it is) then let us add the 'proposed closing' label, so that we could close the issue soon...

**Comment by mattgarrish** [comment]
Everything is done here except for explicit approval of using schema.org as a reference. The vocabulary is now maintained in a CG document and the links to it in the spec have been updated.

The absolute last change is to get the schema.org definitions updated so they link to the new vocabulary document. Our PR for that has been accepted, so we're just waiting on a refresh of the schema.org site to make the changes live.

**Comment by iherman** [comment]
👍

I leave it to our fearless chairs whether we could/should this issue now, or wait for schema.org to get its acts together...

**Comment by jasonjgw** [comment]
In my experience, the issue is typically closed once the final commit addressing it has been merged into the main branch of the document to which it applies - in this case, EPUB Accessibility.

If no more changes are needed, then I think the issue is concluded.

**Comment by mattgarrish** [comment]
> If no more changes are needed, then I think the issue is concluded.

Agree. If it's determined we can't use the reference for some reason in the future, that's a new issue.
